rust/main/chains/hyperlane-sealevel/src/mailbox.rs

```diff
@@ -733,20 +687,180 @@ impl SealevelMailboxIndexer {
         let hyperlane_message =
             HyperlaneMessage::read_from(&mut &dispatched_message_account.encoded_message[..])?;
 
+        let block = self
```

Comment:
just wanna make sure we can rely on this to always give populated data? just coming from the Cosmos stuff, wanna make sure this doesn't get pruned out like it does on Cosmos

Reply:
We'll need to make sure that RPC endpoint we are using contains archived blocks. It is up to RPC endpoint operation to configure the pruning.